googledrive.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/python
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Google Drive Quickstart in Python.
+
+This script uploads a single file to Google Drive.
+"""
+
+from __future__ import print_function
+
+import os
+import os.path
+import io
+import six
+import googleapiclient.http
+import httplib2
+import oauth2client.client
+from storagedrive import StorageDrive
+from pathlib import Path
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = ['https://www.googleapis.com/auth/drive.metadata.readonly']
+# If modifying these scopes, delete the file token.json.
+SCOPES1 = ['https://www.googleapis.com/auth/drive.readonly']
+
+
+class GoogleDrive(StorageDrive):
+
+    def upload(self, file_name):
+
+        # OAuth 2.0 scope that will be authorized.
+        # Check https://developers.google.com/drive/scopes for all available scopes.
+        OAUTH2_SCOPE = 'https://www.googleapis.com/auth/drive'
+        # Location of the client secrets.
+        CLIENT_SECRETS = 'credentials.json'
+        # Path to the file to upload.
+        FILENAME = file_name
+        # Metadata about the file.
+        MIMETYPE = 'text/plain'
+        TITLE = os.path.basename(file_name)
+        DESCRIPTION = f'A text document about {Path(file_name).stem}.'
+        # Perform OAuth2.0 authorization flow.
+        flow = oauth2client.client.flow_from_clientsecrets(
```

Comment:
Shouldn't the variables needed for initializing `client` reside in the constructor of the class.